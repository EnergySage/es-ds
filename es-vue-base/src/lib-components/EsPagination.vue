<template>
    <b-pagination
        v-bind="$attrs"
        class="mb-0"
        first-number
        last-number
        page-class="page-number font-size-sm mx-25 mx-sm-50"
        ellipsis-class="ellipses mx-25 mx-sm-50"
        prev-class="prev-next"
        next-class="prev-next"
        :aria-controls="listID"
        v-on="$listeners">
        <template #prev-text>
            <IconChevronLeft />
        </template>
        <template #next-text>
            <IconChevronRight />
        </template>
    </b-pagination>
</template>

<script>
import { BPagination } from 'bootstrap-vue';
import IconChevronLeft from '../lib-icons/icon-chevron-left.vue';
import IconChevronRight from '../lib-icons/icon-chevron-right.vue';

export default {
    name: 'EsPagination',
    components: {
        BPagination,
        IconChevronLeft,
        IconChevronRight,
    },
    props: {
        listId: {
            type: String,
            required: true,
        },
    },
};
</script>

<style lang="scss">
@import '~@energysage/es-bs-base/scss/includes';

.pagination {
    .page-number, .prev-next, .ellipses {
        .page-link {
            background-color: transparent;
            color: $dark;
            border: 0;
        }
    }

    .page-number {
        display: block !important; // override number limit on mobile
        .page-link {
            height: 2.25rem;
            width: 2.25rem;

            &:hover {
                background-color: transparent;
                color: $black;
                font-weight: bold;
            }
        }

        &.active {
            .page-link {
                background: linear-gradient($white, $white) padding-box,
                    linear-gradient(to bottom right, #f6b140, #c84486) border-box;
                border: 2px solid transparent;
                border-radius: 50em;
                color: $black;
                font-weight: bold;
                padding: 0;
            }
        }
    }

    .prev-next {
        &.disabled {
            .page-link {
                svg {
                    color: $gray-500;
                }
            }
        }

        &:not(.disabled):hover {
            .page-link {
                svg {
                    fill: $black;
                    stroke: $black;
                    stroke-width: 1;
                }
            }
        }

        .page-link {
            padding-left: 0;
            padding-right: 0;

            @include media-breakpoint-up(sm) {
                padding-left: 0.75rem;
                padding-right: 0.75rem;
            }
        }
    }

    .ellipses {
        display: block !important; // override hide ellipses on mobile
        .page-link {
            text-align: center;
        }
    }
}
</style>
