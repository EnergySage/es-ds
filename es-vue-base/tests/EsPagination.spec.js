import { mount } from '@vue/test-utils';
import EsPagination from '@/src/lib-components/EsPagination.vue';
import jestVue from '@/tests/jest.vue.config';

describe('EsPagination', () => {
    // Basic test; should exist for most components
    test('<EsPagination />', async () => {
        const wrapper = mount(EsPagination, {
            ...jestVue,
            template: `
                <ul id="paginated-list">
                    <li>Item 1</li>
                    <li>Item 2</li>
                </ul>
                <EsPagination
                v-model="1"
                :total-rows="2"
                :per-page="1"
                align="center"
                list-id="paginated-list" />
            `,
            components: {
                EsPagination,
            },
        });
        const a11y = await axe(wrapper.element);

        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
        expect(a11y).toHaveNoViolations();
    });
});
