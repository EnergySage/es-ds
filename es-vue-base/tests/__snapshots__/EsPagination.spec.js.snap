// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EsPagination <EsPagination /> 1`] = `
"<ul role="menubar" aria-disabled="false" aria-label="Pagination" class="pagination mb-0 b-pagination">
  <!---->
  <li role="presentation" aria-hidden="true" class="page-item disabled prev-next"><span role="menuitem" aria-label="Go to previous page" aria-disabled="true" class="page-link"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" style="height: 24px; width: 24px;"><path fill-rule="evenodd" clip-rule="evenodd" d="M15.707 3.293a1 1 0 0 1 0 1.414L8.414 12l7.293 7.293a1 1 0 0 1-1.414 1.414l-8-8a1 1 0 0 1 0-1.414l8-8a1 1 0 0 1 1.414 0Z"></path></svg></span></li>
  <!---->
  <!---->
  <li role="presentation" class="page-item page-number font-size-sm mx-sm-50"><button role="menuitemradio" type="button" aria-label="Go to page 1" aria-checked="false" aria-posinset="1" aria-setsize="1" tabindex="0" class="page-link">1</button></li>
  <!---->
  <!---->
  <li role="presentation" aria-hidden="true" class="page-item disabled prev-next"><span role="menuitem" aria-label="Go to next page" aria-disabled="true" class="page-link"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" style="height: 24px; width: 24px;"><path fill-rule="evenodd" clip-rule="evenodd" d="M7.293 3.293a1 1 0 0 1 1.414 0l8 8a1 1 0 0 1 0 1.414l-8 8a1 1 0 0 1-1.414-1.414L14.586 12 7.293 4.707a1 1 0 0 1 0-1.414Z"></path></svg></span></li>
  <!---->
</ul>"
`;
